script/tool/lib/src/version_check_command.dart

```diff
@@ -140,11 +155,26 @@ class VersionCheckCommand extends PackageLoopingCommand {
 
     final List<String> errors = <String>[];
 
-    if (!await _hasValidVersionChange(package, pubspec: pubspec)) {
-      errors.add('Disallowed version change.');
+    bool versionChanged = false;
+    final _CurrentVersionState versionState =
+        await _getVersionState(package, pubspec: pubspec);
+    switch (versionState) {
+      case _CurrentVersionState.unchanged:
+        break;
+      case _CurrentVersionState.validChange:
+        versionChanged = true;
+        break;
+      case _CurrentVersionState.invalidChange:
+        versionChanged = true;
+        errors.add('Disallowed version change.');
+        break;
+      case _CurrentVersionState.unknown:
+        errors.add('Unable to determine previous version.');
+        break;
     }
 
-    if (!(await _hasConsistentVersion(package, pubspec: pubspec))) {
+    if (!(await _hasConsistentVersion(package,
```

Comment:
Should we be calling hasConsistentVersion when _CurrentVersionState.unknown?

Reply:
In general, my experience has been that unless it's expensive (which isn't the case here) it's much better to have CI checks fail but continue, so that we find more errors per run. It's really frustrating to fix a CI failure, push a PR, and wait for checks to run again, only to discover that there's another error now that could have been found before but wasn't. That's a pattern I've been following in most of the tool overhaul work, and it's already made things better.

In this case, errors in the CHANGELOG are almost entirely unrelated to whether the version actually changed (only the one edge case this test now covers); almost all of the time a failure here will just be due to a mismatch between what's in pubspec.yaml and what's in CHANGELOG.md, so surfacing that (so the PR author can fix it in parallel) even if something weird happens with the version check is useful.